### PR TITLE
Fix implode calls

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix syntax. Doc: http://php.net/manual/en/function.implode.php
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11706)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Any idea about how to tests?

**Review by PierreRambaud — APPROVED** [review]

**Comment by MathiasReker** [comment]
I am not able to make a test case, I just follow the syntax

**Comment by MathiasReker** [comment]
@PierreRambaud You could compare the old syntax with the new syntax and check if they output the same. The test could be done in a test environment:)

**Comment by PierreRambaud** [comment]
In fact, I don't know how it was able to work when I read the previous code, that's why I'm afraid to merge without QA validation :(

**Comment by MathiasReker** [comment]
@PierreRambaud I fully agree!

**Comment by MathiasReker** [comment]
@PierreRambaud I make a test case :)

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
> In fact, I don't know how it was able to work when I read the previous code, that's why I'm afraid to merge without QA validation :(

As stated by PHP docs:
> implode() can, for historical reasons, accept its parameters in either order. For consistency with explode(), however, it may be less confusing to use the documented order of arguments.

After reviewing, I think that this can be merged without QA as it doesn't introduce any functional changes.

File: src/Core/Localization/Number/Formatter.php [hunk 1]
```diff
@@ -206,7 +206,7 @@ protected function splitMajorGroups($majorDigits)
             // Reverse back the digits and the groups
             $groups = array_reverse($groups);
             foreach ($groups as &$group) {
-                $group = implode(array_reverse($group));
+                $group = implode('', array_reverse($group));
```
Comment (eternoendless) [inline src/Core/Localization/Number/Formatter.php:209]: This change is unneeded, per doc.

**Comment by MathiasReker** [comment]
The output is the same:
```
foo; bar; qux
foo; bar; qux
```

Check below:
```
<?php

class Foo
{
    protected $title;

    public function __construct($title)
    {
        $this->title = $title;
    }

    public function __toString()
    {
        return $this->title;
    }
}

$array = [
    new Foo('foo'),
    new Foo('bar'),
    new Foo('qux'),
];

echo implode('; ', $array)
."<br>".
implode($array, '; ');
```

File: src/Core/Localization/Number/Formatter.php (hunk 1 above)
Comment (MathiasReker) [reply src/Core/Localization/Number/Formatter.php:209]: Agree, but isn't it better practice to use both parameters?

Comment (MathiasReker) [reply src/Core/Localization/Number/Formatter.php:209]: Should I revert the change?

Comment (eternoendless) [reply src/Core/Localization/Number/Formatter.php:209]: Don't worry about it

**Comment by eternoendless** [comment]
Thank you @MathiasReker

**Pull request merged**